### PR TITLE
Fix json output format and add output tests.

### PR DESCRIPTION
**Pull request by siarheishut**

### CODE REVIEW

File: lib/json/src/json.cpp [hunk 1]
```diff
@@ -1,8 +1,10 @@
 #include "json.h"
 
+#include <cmath>
```
Comment (foxfrikses) [inline lib/json/src/json.cpp:3]: Why do you need `cmath`?

File: lib/json/tests/write_test.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  std::vector<TestCase> test_cases{
+      TestCase{
+          .name = "Monostate",
+          .input = std::monostate{},
+          .want = "null",
+      },
+  };
+
+  Compare(test_cases);
+}
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:141]: Empty line

File: lib/json/tests/write_test.cpp [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+TEST(TestWriteFucntions, TestDouble) {
+  using namespace json;
+
+  std::vector<TestCase> test_cases{
+      TestCase{
+          .name = "Double",
+          .input = 15.123551362,
+          .want = "15.1236",
+      },
+  };
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:124]: Please add a few more to test that it writes only 6 numbers. And add with numbers more than $10^7 - 1$ too please

File: lib/json/tests/write_test.cpp [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+TEST(TestWriteFucntions, TestInt) {
+  using namespace json;
+
+  std::vector<TestCase> test_cases{
+      TestCase{
+          .name = "Int",
+          .input = 1512,
+          .want = "1512",
+      },
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:109]: negative and zero

File: lib/json/tests/write_test.cpp [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+TEST(TestWriteFucntions, TestString) {
+  using namespace json;
+
+  std::vector<TestCase> test_cases{
+      TestCase{
+          .name = "Empty string",
+          .input = "",
+          .want = "\"\"",
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:70]: String with just spaces inside

File: lib/json/tests/write_test.cpp [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  std::vector<TestCase> test_cases{
+      TestCase{
+          .name = "Empty map",
+          .input = Dict{},
+          .want = "{}",
+      },
+      TestCase{
+          .name = "Non-empty map",
+          .input = Dict{{"key 1", 3}, {"key 2", 2}, {"key 3", 1}},
+          .want = "{\"key 1\":3,\"key 2\":2,\"key 3\":1}",
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:56]: Add a value of every type here please, including Dict

File: lib/json/tests/write_test.cpp [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+  std::vector<TestCase> test_cases{
+      TestCase{
+          .name = "Empty array",
+          .input = List{},
+          .want = "[]",
+      },
+      TestCase{
+          .name = "Non-empty array",
+          .input = List{12, 14, 2.5, "what"},
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:36]: Add a value of every type here please, including list

**Review by foxfrikses — CHANGES_REQUESTED** [review]

File: lib/json/src/json.cpp (hunk 1 above)
Comment (siarheishut) [reply lib/json/src/json.cpp:3]: Function mode is in ```cmath```.

File: lib/json/tests/write_test.cpp (hunk 2 above)
Comment (siarheishut) [reply lib/json/tests/write_test.cpp:141]: Done.

File: lib/json/tests/write_test.cpp (hunk 3 above)
Comment (siarheishut) [reply lib/json/tests/write_test.cpp:124]: Done.

File: lib/json/tests/write_test.cpp (hunk 4 above)
Comment (siarheishut) [reply lib/json/tests/write_test.cpp:109]: Done.

File: lib/json/tests/write_test.cpp (hunk 6 above)
Comment (siarheishut) [reply lib/json/tests/write_test.cpp:56]: Done.

File: lib/json/tests/write_test.cpp (hunk 7 above)
Comment (siarheishut) [reply lib/json/tests/write_test.cpp:36]: Done.

File: lib/json/src/json.cpp [hunk 8]
```diff
@@ -172,7 +178,8 @@ void Write(std::ostream &out, const json::Dict &list) {
   for (auto &[key, value] : list) {
     if (!first) out << ",";
     first = false;
-    out << key << ":";
+    Write(out, key);
+    out << ":";
     Write(out, value);
```
Comment (foxfrikses) [inline lib/json/src/json.cpp:183]: `out << ':' << value;`

File: lib/json/tests/write_test.cpp [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+      TestCase{
+          .name = "Empty array",
+          .input = List{},
+          .want = "[]",
+      },
+      TestCase{
+          .name = "Non-empty array",
+          .input = List{12, 41.7, "what", std::monostate{},
+                        Dict{{"key", List{"value", 4}}}},
+          .want = "[12,41.7,\"what\",null,{\"key\":[\"value\",4]}]",
```
Comment (foxfrikses) [inline lib/json/tests/write_test.cpp:38]: Add a list value inside a list too

**Review by foxfrikses — CHANGES_REQUESTED** [review]

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**